sdk/search/search-documents/samples/typescript/src/bufferedSender/uploadDocuments/autoFlushSizeBased.ts

```diff
@@ -0,0 +1,106 @@
+ * by default.
+ */
+const endpoint = process.env.SEARCH_API_ENDPOINT || "";
+const apiKey = process.env.SEARCH_API_KEY || "";
+const TEST_INDEX_NAME = "hotel-live-sample-test3";
+
+function getDocumentsArray(size: number): Hotel[] {
+  const array: Hotel[] = [];
+  for (let i = 1; i <= size; i++) {
+    array.push({
+      hotelId: `${i}`,
+      description:
+        "Best hotel in town if you like luxury hotels. They have an amazing infinity pool, a spa, " +
+        "and a really helpful concierge. The location is perfect -- right downtown, close to all " +
+        "the tourist attractions. We highly recommend this hotel.",
+      descriptionFr:
+        "Meilleur hôtel en ville si vous aimez les hôtels de luxe. Ils ont une magnifique piscine " +
+        "à débordement, un spa et un concierge très utile. L'emplacement est parfait – en plein " +
+        "centre, à proximité de toutes les attractions touristiques. Nous recommandons fortement " +
+        "cet hôtel.",
+      hotelName: "Fancy Stay",
+      category: "Luxury",
+      tags: ["pool", "view", "wifi", "concierge"],
+      parkingIncluded: false,
+      lastRenovationDate: new Date(2010, 5, 27),
+      rating: 5,
+      location: new GeographyPoint(47.678581, -122.131577)
+    });
+  }
+  return array;
+}
+
+export async function main() {
+  console.log(`Running SearchIndexingBufferedSender-uploadDocuments-With Auto Flush Sizes Sample`);
+
+  const credential = new AzureKeyCredential(apiKey);
+  const searchClient: SearchClient<Hotel> = new SearchClient<Hotel>(
+    endpoint,
+    TEST_INDEX_NAME,
+    credential
+  );
+  const indexClient: SearchIndexClient = new SearchIndexClient(endpoint, credential);
+
+  await createIndex(indexClient, TEST_INDEX_NAME);
+  await delay(WAIT_TIME);
+
+  const bufferedClient: SearchIndexingBufferedSender<Hotel> = searchClient.getSearchIndexingBufferedSenderInstance(
+    {
+      autoFlush: true
+    }
+  );
+
+  bufferedClient.on("batchAdded", (response: any) => {
+    console.log("Batch Added Event has been receieved....");
+  });
+
+  bufferedClient.on("batchSent", (response: any) => {
+    console.log("Batch Sent Event has been receieved....");
+  });
+
+  bufferedClient.on("batchSucceeded", (response: any) => {
+    console.log("Batch Succeeded Event has been receieved....");
+    console.log(response);
+  });
+
+  bufferedClient.on("batchFailed", (response: any) => {
+    console.log("Batch Failed Event has been receieved....");
+    console.log(response);
+  });
+
+  const documents: Hotel[] = getDocumentsArray(1001);
+  bufferedClient.uploadDocuments(documents);
+
+  await WAIT_TIME;
+
+  let count = await searchClient.getDocumentsCount();
+  while (count !== documents.length) {
+    await delay(WAIT_TIME);
+    count = await searchClient.getDocumentsCount();
+  }
+
+  // When the autoFlush is set to true, the user
+  // has to call the dispose method to clear the
+  // timer.
+  bufferedClient.dispose();
```

Comment:
```Should it also call flush if there are unsent items?```
I think it makes sense. Added it.

```should dispose be something you have to await?```
Since we are calling a flush in dispose, I am adding an await here. 

Resolved